skore/src/skore/utils/_progress_bar.py

```diff
@@ -0,0 +1,84 @@
+from functools import wraps
+
+from rich.progress import (
+    BarColumn,
+    Progress,
+    SpinnerColumn,
+    TextColumn,
+)
+
+
+def create_progress_bar():
+    """Create a consistent progress bar style."""
+    from skore import console  # avoid circular import
+
+    return Progress(
+        SpinnerColumn(),
+        TextColumn("[bold cyan]{task.description}"),
+        BarColumn(
+            complete_style="dark_orange",
+            finished_style="dark_orange",
+            pulse_style="orange1",
+        ),
+        TextColumn("[orange1]{task.percentage:>3.0f}%"),
+        console=console,
+        expand=False,
+    )
+
+
+class ProgressManager:
+    _instance = None
+    _progress = None
+
+    @classmethod
+    def get_progress(cls):
+        if cls._progress is None:
+            cls._progress = create_progress_bar()
+            cls._progress.start()
+        return cls._progress
+
+    @classmethod
+    def stop_progress(cls):
+        if cls._progress is not None:
+            try:
+                cls._progress.stop()
+            finally:
+                cls._progress = None
+
+
+class ProgressDecorator:
```

Comment:
This should be a function

Reply:
@augustebaum you need to give more info.

FYI, I don't like the current design of the progress bar. I need to manually stop the thread manager and I need some private sentinel in the class to know if I should add a progress bar to another one to have nested progress bar. It is really invasive but it is a working POC.

So if you think that a design with a function can simplify and answer the different constraints and really looking forward. But you need to be explicit in how you see the design of this function, what problem does it solve compare to the actual design.

Reply:
> after all I'm reviewing a draft PR ^^

Always good to have feedback soonish ;)

> in its current state, a class containing nothing but a __call__ method is equivalent to a simple function, so it shouldn't be a class.

Yep I agree with this point. I think that the `description` can be passed with context manager as a function as well.

As said before, I don't like the design. I'm not sure that a decorator is great indeed.

Reply:
Resolving since we made it better.